### PR TITLE
Making HashCodeCombiner usage non-fluent

### PR DESCRIPTION
**Pull request by pranavkm**

### CODE REVIEW

**Comment by pranavkm** [comment]
cc @ajaybhargavb

**Comment by pranavkm** [comment]
Reaction to https://github.com/aspnet/Common/issues/40

File: src/Microsoft.AspNet.Razor.Test.Sources/TagHelperAttributeDescriptorComparer.cs [hunk 1]
```diff
@@ -38,15 +38,17 @@ public bool Equals(TagHelperAttributeDescriptor descriptorX, TagHelperAttributeD
...
-                .Add(descriptor.IsIndexer)
-                .Add(descriptor.Name, StringComparer.Ordinal)
-                .Add(descriptor.PropertyName, StringComparer.Ordinal)
-                .Add(descriptor.TypeName, StringComparer.Ordinal)
-                .Add(descriptor.IsStringProperty)
-                .Add(TagHelperAttributeDesignTimeDescriptorComparer.Default.GetHashCode(
-                    descriptor.DesignTimeDescriptor))
-                .CombinedHash;
+            var hashCodeCombiner = HashCodeCombiner.Start();
+
```
Comment (dougbu) [inline src/Microsoft.AspNet.Razor.Test.Sources/TagHelperAttributeDescriptorComparer.cs]: Suggest consistency in these `GetHashCode()` implementations would be nice. Looks like w/o the blank line is more common.

File: src/Microsoft.AspNet.Razor/Parser/SyntaxTree/Block.cs [hunk 2]
```diff
@@ -222,7 +223,8 @@ public override bool EquivalentTo(SyntaxTreeNode node)
 
         public override int GetEquivalenceHash()
         {
-            var combiner = HashCodeCombiner.Start().Add(Type);
+            var combiner = HashCodeCombiner.Start();
```
Comment (dougbu) [inline src/Microsoft.AspNet.Razor/Parser/SyntaxTree/Block.cs]: `hashCodeCombiner` for consistency

**Comment by dougbu** [comment]
:shipit:

**Comment by ajaybhargavb** [comment]
:shipit:

**Pull request closed without merging**